### PR TITLE
Add timezone-aware transaction querying

### PR DESCRIPTION
**Pull request by sergeyklay**
### 🎯 Scope & Context

**Type:** Fix

**Intent:** Resolve timezone boundary issues with monthly transaction querying by implementing a timezone-aware two-phase approach. Payments logged at month boundaries (e.g., Oct 1 00:00 in UTC+2 = Sept 30 22:00 UTC) are now correctly attributed to the intended month regardless of server timezone.

### 🧭 Reviewer Guide

**Complexity:** High

#### Entry Point

Start with `lib/services/TransactionService.ts` to understand the core timezone logic. The file begins with detailed helper functions (`calculateMonthBoundaries`, `calculateExtendedQueryBoundaries`, `calculateFilterBoundaries`, and `isPaymentInMonth`) that implement a two-phase boundary approach: extended query boundaries to fetch transactions across all timezone offsets (UTC-12 to UTC+14), followed by in-memory filtering to identify payments truly within the target month. This logic is applied to four transaction query methods: `getByBillIdAndMonth`, `getPaymentsByMonth`, `getMonthlyPaymentTotals`, and `getPaymentsByYearAggregatedByBill`. Then review `lib/services/TransactionService.test.ts` to validate the boundary handling across edge cases (month/year transitions, boundary inclusions/exclusions).

#### Sensitive Areas

- `lib/services/TransactionService.ts`: Timezone boundary calculation logic (`calculateFilterBoundaries`) and post-query filtering (`isPaymentInMonth`) that determines which transactions belong to each month - this affects all transaction aggregations and historical views
- `lib/services/TransactionService.ts`: Input validation for month/year parameters to prevent invalid queries
- `lib/services/TransactionService.test.ts`: Timezone boundary regression tests covering UTC+2 scenarios, month/year boundaries, and adjacent-month exclusions - validates correct filtering behavior

### ⚠️ Risk Assessment

- **Breaking Changes:** No breaking changes - all public method signatures and visibility remain unchanged; internal helper functions are not exported
- **Migrations/State:** No migrations or state changes required

### CODE REVIEW

File: lib/services/TransactionService.ts [hunk 1]
```diff
@@ -176,39 +324,61 @@ export const TransactionService = {
     month: number,
     year: number
   ): Promise<Transaction[]> {
-    const monthDate = new Date(year, month - 1, 1);
-    const monthStart = startOfMonth(monthDate);
-    const monthEnd = endOfMonth(monthDate);
+    if (month < 1 || month > 12 || isNaN(year)) {
+      return [];
+    }
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply lib/services/TransactionService.ts]: This should be fixed now. Please check
